### PR TITLE
fix(pill-selector-dropdown): update suggested pills filter

### PR DESCRIPTION
**Pull request by nickgriffin1**
Problem: this was build under the assumption that all pills contained an "id" value, which is not true. All pill do have a "value". Value is typically either an "id" or an "email" so we can filter the list of suggested pills by both.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @nickgriffin1 has signed the CLA. Thanks for the pull request!

File: src/components/pill-selector-dropdown/SuggestedPillsRow.js [hunk 1]
```diff
@@ -9,19 +9,21 @@ import './SuggestedPillsRow.scss';
...
-    selectedPillsIDs = [],
+    selectedPillsValues = [],
     suggestedPillsData = [],
     title,
 }: Props) => {
-    // Prevents pills from being rendered that are in the form
-    const filteredSuggestedPillData = suggestedPillsData.filter(item => !selectedPillsIDs.includes(item.id));
+    // Prevents pills from being rendered that are in the form by checking for value which is either an ID or email
+    const filteredSuggestedPillData = suggestedPillsData.filter(
+        item => !(selectedPillsValues.includes(item.email) || selectedPillsValues.includes(item.id)),
```
Comment (priyajeet) [inline src/components/pill-selector-dropdown/SuggestedPillsRow.js]: Just double check to make sure Array `includes` is polyfilled

Comment (priyajeet) [reply src/components/pill-selector-dropdown/SuggestedPillsRow.js]: Can "email" and "id" be masked somehow? As in, you provide what to filter against.

Comment (nickgriffin1) [reply src/components/pill-selector-dropdown/SuggestedPillsRow.js]: Tested as a linked dependency on IE11/Win10 and no console error thrown. I feel like we should normally be able to assume that is working right?

In this case keyProp would be a prop passed from PillSelectorDropdown? That seems like reasonable solution that is a bit more reusable to cover other use cases. I'll change this.

Comment (nickgriffin1) [reply src/components/pill-selector-dropdown/SuggestedPillsRow.js]: I'll merge this for now and Think about how best to implement this prop in a future commit.

**Pull request merged**